migrations/2024-02-20-093804_ai_help_metadata/up.sql

```diff
@@ -0,0 +1,30 @@
+CREATE TYPE ai_help_message_status AS ENUM (
+    'success',
+    'search_error',
+    'ai_api_error',
+    'completion_error',
+    'moderation_error',
+    'no_user_prompt_error',
+    'token_limit_error',
+    'user_stopped',
+    'user_timeout',
+    'unknown'
+);
+
+CREATE TABLE ai_help_message_meta (
+    id                  BIGSERIAL PRIMARY KEY,
+    user_id             BIGSERIAL REFERENCES users (id) ON DELETE CASCADE,
+    chat_id             UUID NOT NULL,
+    message_id          UUID NOT NULL,
+    parent_id           UUID DEFAULT NULL REFERENCES ai_help_message_meta (message_id) ON DELETE CASCADE,
```

Comment:
Wouldn't this cause a database error if a user asks a follow-up question where the parent message was created before we land this feature?

Reply:
I think we should simply remove the foreign key.

Reply:
Fine with that.